library/core/resource-loader/src/main/java/org/quiltmc/qsl/resource/loader/mixin/DefaultResourcePackMixin.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.mixin;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.function.Predicate;
+
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.resource.DefaultResourcePack;
+import net.minecraft.resource.ResourceType;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.resource.loader.impl.ModNioResourcePack;
+import org.quiltmc.qsl.resource.loader.impl.ResourceLoaderImpl;
+
+@Mixin(DefaultResourcePack.class)
+public abstract class DefaultResourcePackMixin {
+	// Redirects all resource access to the MC resource pack.
+	@Unique
+	final ModNioResourcePack internalPack = this.locateAndLoad();
+
+	@SuppressWarnings({"ConstantConditions", "EqualsBetweenInconvertibleTypes"})
+	@Unique
+	private ModNioResourcePack locateAndLoad() {
+		return ResourceLoaderImpl.locateAndLoadDefaultResourcePack(
+				this.getClass().equals(DefaultResourcePack.class) ?
+						ResourceType.SERVER_DATA : ResourceType.CLIENT_RESOURCES
+		);
+	}
+
+	/**
+	 * @author QuiltMC
```

Comment:
Author should refer to the code author(s) probably and not quiltmc

Reply:
In the mixin overwrite context it mostly refers to who provide the mixin imo.

Reply:
So two conflicting overwrites will note the authors?

Reply:
I'm pretty sure it does?

mcdev warns about it needing an author and reason fields in the doc at least.